### DISCUSSION TITLE
IP of trace instructions

### DISCUSSION THREAD

**Discussion by SteliosKaragiorgis** [ChampSim/ChampSim#459] [Q&A] [answered]
Hello. So i have a question regarding the IP of the instructions. I print the instruction in the format:

IP(hex): 0x47f721
Is Branch: 1
Branch Taken: 0
Branch Type: BRANCH CONDITIONAL

IP(hex): 0x47f727
Is Branch: 0
Branch Taken: 0
Branch Type: NOT BRANCH

IP(hex): 0x47f72d
Is Branch: 0
Branch Taken: 0
Branch Type: NOT BRANCH

IP(hex): 0x47f733
Is Branch: 0
Branch Taken: 0
Branch Type: NOT BRANCH

IP(hex): 0x47f738
Is Branch: 0
Branch Taken: 0
Branch Type: NOT BRANCH

IP(hex): 0x47f73d
Is Branch: 0
Branch Taken: 0
Branch Type: NOT BRANCH

IP(hex): 0x47f744
Is Branch: 0
Branch Taken: 0
Branch Type: NOT BRANCH

IP(hex): 0x47f746
Is Branch: 0
Branch Taken: 0
Branch Type: NOT BRANCH

IP(hex): 0x47f749
Is Branch: 0
Branch Taken: 0
Branch Type: NOT BRANCH

IP(hex): 0x47f74c
Is Branch: 1
Branch Taken: 1
Branch Type: BRANCH CONDITIONAL

So my question is: Why the IP is not increament in constant number.
For example on MIPS ISA the PC is increment +4 for every instruction.
I run the trace `/400.perlbench-41B.champsimtrace.xz` from the website https://dpc3.compas.cs.stonybrook.edu/champsim-traces/speccpu/

What ISA is this trace file?
Thanks in advance

**Comment by ngober** [comment]
There are many different instruction set architectures, and ChampSim does not make assumptions about which it is modeling. It appears that the trace in question was generated with a variable-length instruction set. If your research requires a fixed-length instruction set, you may need to generate your own traces, rather than use some provided by someone else.

**Reply by SteliosKaragiorgis** [reply]
Thank you for your answer. Is it a way to understand what ISA is each trace file? 
For example this trace file if it is x86

**Reply by ngober** [reply, marked as answer]
You would need to ask the person who generated the trace. That information is currently not available in the trace (I will continue to point people at #382).

**Reply by SteliosKaragiorgis** [reply]
Thank you for the help
